LineItem properties cast and parse

Cyberlane opened the pull request:
When setting the line item properties `LineItemVatAmount` and `LineItemVatPercentage`, if you call `Cart.AcceptChanges()`, the values will be saved to the database and the values are automatically cast into strings. When trying to cast a string to a decimal, an invalid cast exception will be thrown. Since the value cannot be trusted to remain unchanged, a change is needed to handle both casting and parsing.

The change I have sent will first attempt to safely cast to a decimal, and failing that, attempt to parse.

asbjornu commented:
@Cyberlane Now that we have implemented test automation on AppVeyor, can you please rebase this PR on current `master` so we can see if any tests fail? It would also be good to have a test for the problem you've discovered, so it would be really great if you could add that as well. Thanks!

Cyberlane commented:
@asbjornu sorry for the late reply, I was away on semester/vacation.

I haven't actually used this library for EPiServer since December 2015, and now use my own so I am uncertain if the issues on my original PR still apply. I imagine a good number of changes have happened to the library, perhaps addressing all the issues I had faced. Would you like me to simply close my outstanding PRs?

asbjornu commented:
@Cyberlane: I'm not sure whether your PR still apply or not, but I'm glad to inform that this module is going to receive a lot more attention going forward and we would love to see more contributions from you and others in the future. Please don't hesitate to get in touch with us and if you would like to use this module in a future project, that would be great.
